addons/block_code/ui/blocks/block/block.gd

```diff
@@ -61,6 +64,17 @@ func get_instruction_node() -> InstructionTree.TreeNode:
 	return node
 
 
+func update_resources(undo_redo: EditorUndoRedoManager):
+	if resource == null:
+		resource = SerializedBlockTreeNode.new()
+		resource.serialized_block = SerializedBlock.new(get_block_class(), get_serialized_props())
+
+	var serialized_props = get_serialized_props()
+	if serialized_props != resource.serialized_block.serialized_props:
+		undo_redo.add_undo_property(resource.serialized_block, "serialized_props", resource.serialized_block.serialized_props)
+		undo_redo.add_do_property(resource.serialized_block, "serialized_props", serialized_props)
```

Comment:
One thing I thought about this morning with an eye towards the future decoupled data and UI is that maybe we don't want to modify the resources in place. If we want to get to a place where the UI only refers to a code resource, then I think that's easier to handle if the serialized block is treated immutably.

In other words, rather than updating `serialized_props` on an existing `SerializedBlock`, detect if `serialized_props` has changed and replace the `serialized_block` property in that case. Like:

```
if resource == null:
    resource = SerializedBlockTreeNode.new()

if resource.serialized_block == null:
    resource.serialized_block = SerializedBlock.new(get_block_class(), get_serialized_props())
    return

var serialized_props = get_serialized_props()
if serialized_props != resource.serialized_block.serialized_props:
    undo_redo.add_undo_property(resource, "serialized_block", resource.serialized_block)
    undo_redo.add_do_property(resource, "serialized_block", SerializedBlock.new(get_block_class(), get_serialized_props()))
```

WDYT?

Reply:
@dbnicholson hmm, this will always generate a new resource ID for SerializedBlock, which is the opposite this PR is trying to do. What would be exactly the benefit?

Reply:
It's really in between both worlds. If the block inputs haven't changed, leave it as is. If the have, replace the object. The idea being that if there are multiple objects referencing this serialized block, then they may not be prepared to have it modified in place. That's just theoretical, though, as the serialized blocks are always referenced from a single UI block right now. So, let's just move on.